pkg/controller/plan/scheduler/vsphere/scheduler.go

```diff
@@ -117,9 +118,22 @@ func (r *Scheduler) buildInFlight() (err error) {
 	// we need to use the plan from the context rather
 	// than from the list of plans that are retrieved below.
 	for _, vmStatus := range r.Plan.Status.Migration.VMs {
+		if vmStatus.HasCondition(Canceled) {
+			continue
+		}
 		vm := &model.VM{}
 		err = r.Source.Inventory.Find(vm, vmStatus.Ref)
 		if err != nil {
+			if errors.As(err, &web.NotFoundError{}) {
+				vmStatus.SetCondition(libcnd.Condition{
+					Type:     api.ConditionCanceled,
+					Status:   libcnd.True,
+					Category: api.CategoryAdvisory,
+					Reason:   NotFound,
+					Message:  "VM was deleted from inventory.",
```

Comment:
```suggestion
					Message:  "VM was not found in inventory.",
```

we know it's not found and we assume it was deleted, IMHO, best to state what we know and let the user assume it's not found because it was deleted